src/Traits/Http.php

```diff
@@ -253,15 +267,17 @@ protected function prepareMultipartParams(array $params, $inputFileField): array
 
         $inputFile = $params[$inputFileField];
 
-
+        //All files should be provided in an InputFile object
         if (is_resource($inputFile)) {
             throw CouldNotUploadInputFile::resourceShouldBeInputFileEntity($inputFileField);
         }
 
+        //If the user provides a path to a file, attempt to create an InputFile Object automatically for them.
         if (is_string($inputFile)) {
             $params[$inputFileField] = InputFile::create($inputFile);
         }
```

Comment:
`uploadFile` is already checking if it's a file_id and if yes, then does a post request directly. This is if it's not a file_id part. So if it's not a file_id, then it's either a URL or file path. In any case, we need to use InputFile entity to upload such files.